### ISSUE TITLE
Matches is empty if half of cropLength is smaller than the matched query word.

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/MeiliSearch#599]
Should _matchesInfo be empty if the cropping cuts the matched query word?

### Examples
This is a search with the following query parameters: 
```
filters: "title:Le Petit Prince",
attributesToCrop: "*",
cropLength: 10,
matches: true
```

#### With a cropLength of `5`
```json
{
   "hits":[
      {
         "id":456,
         "title":"Le Petit Prince",
         "comment":"A french book about a prince that walks on little cute planets",
         "_formatted":{
            "id":456,
            "title":"etit Princ",
            "comment":"ut a princ"
         },
         "_matchesInfo":{

         }
      }
   ],
   "offset":0,
   "limit":20,
   "processingTimeMs":0,
   "query":"prince"
}
```

Here the mathesInfo object is **empty**.

####  With a cropLength of 6
```json
{
   "hits":[
      {
         "id":456,
         "title":"Le Petit Prince",
         "comment":"A french book about a prince that walks on little cute planets",
         "_formatted":{
            "id":456,
            "title":"Petit Prince",
            "comment":"out a prince"
         },
         "_matchesInfo":{
            "title":[
               {
                  "start":6,
                  "length":6
               }
            ],
            "comment":[
               {
                  "start":6,
                  "length":6
               }
            ]
         }
      }
   ],
   "offset":0,
   "limit":20,
   "processingTimeMs":0,
   "query":"prince"
}
```
Here it is **not empty**.

**Comment by bidoubiwa** [comment]
Highlight is also not applied in the same case.

**Comment by bidoubiwa** [comment]
This has been fixed in the v0.10 somewhere
